src/panels/lovelace/ha-panel-lovelace.ts

```diff
@@ -263,15 +264,20 @@ class LovelacePanel extends LitElement {
 
   private _checkLovelaceConfig(config: LovelaceConfig) {
     // Somehow there can be badges with value null, we remove those
-    config.views.forEach((view) => {
-      if (view.badges) {
-        view.badges = view.badges.filter(Boolean);
+    let checkedConfig;
+    config.views.forEach((view, index) => {
+      if (view.badges && !view.badges.every(Boolean)) {
+        checkedConfig = checkedConfig || { ...config };
+        checkedConfig.views[index] = { ...view };
```

Comment:
Shouldn't we copy `views` here too ?

Reply:
Yes. I have no idea how this could work in my testing... unless arrays are immutable?
Will fix in a few hours, when I'm back at a computer.

Reply:
Maybe your test was not running in strict mode? (although I assume our bundler adds that…)

![image](https://user-images.githubusercontent.com/1444314/74462643-6e645300-4e45-11ea-9f8c-64965dbbc82f.png)

Reply:
Turns out my test was running before the freeze. I'm glad we have this review process...